### DISCUSSION TITLE
Unit testing a react component with NextIntlProvider using Jest

### DISCUSSION THREAD

**Discussion by kylemorena** [amannn/next-intl#513] [Q&A]
Hello everyone,

I'm getting an error working on `"next-intl": "3.0.0-beta.18"`, `"jest": "^27.5.1",`, `"ts-jest": "^27.1.3",`

```
Cannot find module 'use-intl/_useLocale' from 'node_modules/next-intl/dist/development/react-client/useLocale.js'

    Require stack:
      node_modules/next-intl/dist/development/react-client/useLocale.js
      node_modules/next-intl/dist/development/link/Link.js
      node_modules/next-intl/dist/development/react-client/Link.js
      node_modules/next-intl/dist/development/react-client/index.js
      node_modules/next-intl/dist/development/index.js
      node_modules/next-intl/dist/index.js
```
Below here is my code:

useExperienceDuration.test.tsx

```

async function createWrapper(lang: string) {
		const common = await import(`@/locales/${lang}/common.json`);
		return function Wrapper({ children }) {
			return (
				<NextIntlProvider locale={lang} messages={common}>
					{children}
				</NextIntlProvider>
			);
		};
	}


	it("Returns the total of duration giving a range", async () => {
		const { result } = renderHook(
			() => {
				return useExperienceDuration();
			},
			{
				wrapper: await createWrapper("en")
			}
		);
		const label = convertLabel(result.current.labelDuration, "en");
		expect(label).toBe(expected);
	})
```

**Has someone recently faced it, and solved it?**

Many thanks for considering my request.

Kyle

**Comment by amannn** [comment]
If you install `next-intl@3.0.0-beta.18` it should include `use-intl@3.0.0-beta.3` which provides the entry point that seems to cause your issue. Maybe reinstalling dependencies and making sure `use-intl` was added correctly can help to resolve the issue?

**Reply by kylemorena** [reply]
Hello, looking inside my lock file, I got 

```
  /next-intl@3.0.0-beta.18(next@13.4.19)(react@18.2.0):
    resolution: {integrity: sha512-RK3gM4NsA3mwzroMfIEgrCs5JDA/KVrMBzWlk0WThbqSeWy4vULz2FBUEqQdE8dSIdhxsArwGbrZwUNbCgSkWw==}
    peerDependencies:
      next: ^10.0.0 || ^11.0.0 || ^12.0.0 || ^13.0.0
      react: ^16.8.0 || ^17.0.0 || ^18.0.0
    dependencies:
      '@formatjs/intl-localematcher': 0.2.32
      negotiator: 0.6.3
      next: 13.4.19(@babel/core@7.22.20)(react-dom@18.2.0)(react@18.2.0)(sass@1.32.12)
      react: 18.2.0
      server-only: 0.0.1
      use-intl: 3.0.0-beta.3(react@18.2.0)
    dev: false
```

**Reply by amannn** [reply]
Can you share a reproduction?

Works fine for me locally with this `jest.config.js`:

```js
/* eslint-env node */
const nextJest = require('next/jest');

const createJestConfig = nextJest({dir: './'});

module.exports = createJestConfig({
  moduleNameMapper: {
    '^next$': require.resolve('next'),
    '^next/navigation$': require.resolve('next/navigation')
  },
  testEnvironment: 'jsdom',
  rootDir: 'src'
});
```

(tested with [example-next-13](https://github.com/amannn/next-intl/tree/feat/next-13-rsc/examples/example-next-13) on the RSC branch)

**Reply by kylemorena** [reply]
Ok now working was about jest.config.js
Thank you!

**Comment by stevefiedler** [comment]
i am using next-intl@3.1.2 and am still getting this error in my jest unit tests:

    Cannot find module 'use-intl/_useLocale' from 'node_modules/next-intl/dist/development/react-client/useLocale.js'

I am now using NextIntlClientProvider rather than the older IntlProvider.

Maybe this works for App Router and not for Pages Router any more?  That doesn't make sense to me either.

In the stack trace I see:
      node_modules/next-intl/dist/development/react-client/useLocale.js
      node_modules/next-intl/dist/development/index.react-client.js
      node_modules/next-intl/dist/index.react-client.js

Where is next-intl/_useLocale, because I don't see it?

**Reply by stevefiedler** [reply]
Actually I found it in use-intl just where it should be but I'm still getting the error.

Ideas?

**Comment by gorangatuzo** [comment]
I have the same problem running `"next-intl": "^3.3.2",` and `"next": "^14.0.4",`

**Comment by stevefiedler** [comment]
I agree this is still broken.  I am also using the Pages router.  The application works fine, it's just all the tests failing.
